platform/darwin/src/MGLOfflineStorage.h

```diff
@@ -448,6 +450,40 @@ MGL_EXPORT
 
 - (void)putResourceWithUrl:(NSURL *)url data:(NSData *)data modified:(nullable NSDate *)modified expires:(nullable NSDate *)expires etag:(nullable NSString *)etag mustRevalidate:(BOOL)mustRevalidate __attribute__((deprecated("", "-preloadData:forURL:modificationDate:expirationDate:eTag:mustRevalidate:")));
 
+
+/**
+A block to be called once the data  has been preloaded.
+
+ @param url The URL of the data that was pre-loaded.
+ @param error Contains a pointer to an error object (if any) indicating why the
+ data could not be pre-loaded.
+*/
+typedef void (^MGLOfflinePreloadDataCompletionHandler)(NSURL * url, NSError * _Nullable error);
```

Comment:
Nit: It’s more common for typedefs to be declared at the top of the file outside the interface declaration. Declaring it here doesn’t break Clang or jazzy but could be harder to find when looking for typedefs in the usual place.